storage/client/src/main/java/com/azure/storage/file/DirectoryAsyncClient.java

```diff
@@ -3,77 +3,527 @@
+     */
+    public Flux<FileRef> listFilesAndDirectories(String prefix, Integer maxResults) {
+        return azureFileStorageClient.directorys().listFilesAndDirectoriesSegmentWithRestResponseAsync(shareName, directoryName, prefix, shareSnapshot, null, maxResults, null, Context.NONE)
+                  .flatMapMany(response -> nextPageForFileAndDirecotries(response, prefix, maxResults));
     }
 
-    public Flux<HandleItem> getHandles(int maxResult, boolean recursive) {
-        throw new UnsupportedOperationException();
+    /**
+     * List of open handles on a directory or a file.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * <p>Get 10 handles with recursive call.</p>
+     *
+     * <pre>
+     * client.getHandles(10, true)
+     *     .subscribe(handleItem -&gt; System.out.printf("Get handles completed with handle id %s", handleItem.handleId()));
+     * </pre>
+     * @param maxResult Optional. The number of results will return per page
+     * @param recursive Specifies operation should apply to the directory specified in the URI, its files, its subdirectories and their files.
+     * @return {@link HandleItem handles} in the directory that satisfy the requirements
+     */
+    public Flux<HandleItem> getHandles(Integer maxResult, boolean recursive) {
+        return azureFileStorageClient.directorys().listHandlesWithRestResponseAsync(shareName, directoryName, null, maxResult, null, shareSnapshot, recursive, Context.NONE)
+                   .flatMapMany(response -> nextPageForHandles(response, maxResult, recursive));
     }
 
+    /**
+     * Closes a handle or handles opened on a directory or a file at the service. It is intended to be used alongside {@link DirectoryAsyncClient#getHandles(Integer, boolean)} .
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * <p>Force close handles with handles returned by get handles in recursive.</p>
+     *
+     * <pre>
+     * client.getHandles(10, true)
+     *     .subscribe(handleItem -&gt; {
+     *         client.forceCloseHandles(handleItem.handleId(), true).subscribe(numOfClosedHandles -&gt
+     *              System.out.printf("Close %d handles.", numOfClosedHandles)
+     *     )});
+     * </pre>
+     * @param handleId Specifies the handle ID to be closed. Use an asterisk ('*') as a wildcard string to specify all handles.
+     * @param recursive A boolean value that specifies if the operation should also apply to the files and subdirectories of the directory specified in the URI.
+     * @return The counts of number of handles closed
+     */
     public Flux<Integer> forceCloseHandles(String handleId, boolean recursive) {
-        throw new UnsupportedOperationException();
+        return azureFileStorageClient.directorys().forceCloseHandlesWithRestResponseAsync(shareName, directoryName, handleId, null, null, shareSnapshot, recursive, Context.NONE)
```

Comment:
directories, not directorys. I see a few API with this typo, please ensure it is all resolved and spelt correctly.